index.js

```diff
@@ -49,16 +50,13 @@ async function fastifyStatic (fastify, opts) {
   }
 
   const allowedPath = opts.allowedPath
-
-  if (opts.prefix === undefined) opts.prefix = '/'
-
-  let prefix = opts.prefix
+  let prefix = (opts.prefix ??= '/')
```

Comment:
wont you mutate the opts.prefix value with this? i prefer not to mutate it, as opts is an object, so you change it also on the object and not only in your scope

first iteration:
```js
let prefix = opts.prefix ?? '/'

if (
  !opts.prefixAvoidTrailingSlash &&
  prefix.at(-1) !== '/'
) {
  prefix += '/'
}
```

second iteration:


```js
const prefix = opts.prefix === undefined
  ? '/'
  : !opts.prefixAvoidTrailingSlash && prefix.at(-1) !== '/'
    ? prefix + '/'
    : prefix
```

but if this is faster? idk
but maybe something to consider (despite some fastify devs disliking ternary operations)

Reply:
but on the other hand, this is not a perf critical path.

Reply:
Agreed, but I actually kept the [old logic](https://github.com/fastify/fastify-static/pull/405/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L53)

Not mutating causes some errors because some checks were written for the mutated property, a little leaky